extensions/src/main/java/org/pentaho/platform/plugin/services/pluginmgr/PentahoSystemPluginManager.java

```diff
@@ -369,6 +373,18 @@ private void registerPlugin( final IPlatformPlugin plugin ) throws PlatformPlugi
       .get( GenericApplicationContext.class, null, Collections.singletonMap( PLUGIN_ID, plugin.getId() ) );
     createAndRegisterLifecycleListeners( plugin, loader );
 
+    // set the provider that will be used to load the plugin's resource bundles for localization
+    plugin.setResourceBundleProvider( new CachingResourceBundleProvider( ( Locale locale ) -> {
+      Objects.requireNonNull( loader );
+      Objects.requireNonNull( locale );
+
+      var baseName = StringUtil.isEmpty( plugin.getResourceBundleClassName() )
+        ? PlatformPlugin.DEFAULT_RESOURCE_BUNDLE_CLASS_NAME
```

Comment:
See related comment regarding the default resource bundle.

According to that, if there is no resource bundle, then immediately return `null`.